regulations/docket.py

```diff
@@ -7,48 +6,61 @@
 from django.conf import settings
 from django.core.cache import caches
 
-Field = namedtuple('Field', 'max_length, required')
-
 logger = logging.getLogger(__name__)
 
+cache = caches['regs_gov_cache']
+
 
-def get_document_metadata(document_id):
-    """ Retrieve document metadata from regulations.gov. """
-    logger.debug("Getting docket metadata")
-    response = requests.get(
-        settings.REGS_GOV_API_URL,
-        params={'D': document_id},
-        headers={'X-Api-Key': settings.REGS_GOV_API_KEY}
-    )
-    if response.status_code == requests.codes.ok:
-        return response.json()
-    else:
-        logger.error("Failed to lookup regulations.gov: {}",
-                     response.status_code, response.text)
+def fetch(url, **kwargs):
+    kwargs['headers'] = kwargs.get(
+        'headers', {'X-Api-Key': settings.REGS_GOV_API_KEY})
+    response = requests.get(url, **kwargs)
+    if response.status_code != requests.codes.ok:
         response.raise_for_status()
+    return response.json()
 
 
 def get_document_fields(document_id):
     """ Retrieve the field list from the document metadata.
         Use a cache as the data hardly ever changes.
         We ignore the 'general_comment' field as it has special treatment.
     """
-    cache = caches['regs_gov_cache']
-    cache_key = document_id
-    fields = cache.get(cache_key)
-
-    if fields is None:
-        document_metadata = get_document_metadata(document_id)
-        fields = {
-            field['attributeName']: Field(field['maxLength'],
-                                          field['required'])
-            for field in document_metadata['fieldList']
-            if field['attributeName'] != 'general_comment'
-        }
-        cache.set(cache_key, fields)
+    fields = cache.get(document_id)
+    if fields is not None:
+        return fields
+
+    metadata = fetch(settings.REGS_GOV_API_URL, params={'D': document_id})
+    fields = {
+        field['attributeName']: field
+        for field in metadata['fieldList']
+        if field['attributeName'] != 'general_comment'
+    }
+
+    add_options(fields)
+    add_dependent_options(fields)
+
+    cache.set(document_id, fields)
     return fields
 
 
+def add_options(fields):
```

Comment:
I found `options` confusing and would prefer a name like domain/domain_values/legal_values. It may just be me, please feel free to ignore.